src/server.c

```diff
@@ -874,11 +878,11 @@ struct redisCommand redisCommandTable[] = {
     /* EVAL can modify the dataset, however it is not flagged as a write
      * command since we do the check while running commands from Lua. */
     {"eval",evalCommand,-3,
-     "no-script @scripting",
+     "no-script can-replicate @scripting",
```

Comment:
i guess that SCRIPT LOAD, and PUBLISH commands needs that too.
basically any command that calls forceCommandPropagation.
maybe we even want to add an assertion on c->cmd in forceCommandPropagation?

Reply:
Can you explain the point of the assertion in forceCommandPropagation that you proposed?

Reply:
to make sure that we don't add a some future command that uses forceCommandPropagation and isn't marked with either `write` or  `can-replicate`.
maybe we should also have such assertion in the call to `propagate()` in `call()`.

Reply:
Oh, that assert seems reasonable. I thought about adding one in the propagate path, but there are ways you can get there like multi-exec'ing a client pause + a write. I throw a warning there now, which isn't great.